test/integration/crud/explain.test.ts

```diff
@@ -77,21 +77,24 @@ describe('Explain', function () {
   });
 
   context('when explain is set to false', () => {
-    it('only queryPlanner property is used in explain result', async function () {
+    it('deleteOne only returns queryPlanner property in explain result', async function () {
       const explanation = await collection.deleteOne({ a: 1 }, { explain: false });
       expect(explanation).property('queryPlanner').to.exist;
+      expect(explanation).to.not.have.property('executionStats');
```

Comment:
If this is the differentiating assertion for `explain: false` vs `explain: true`, I would expect the `explain: true` results to assert that this property does exist; however this still doesn't guarantee that ONLY `queryPlanner` is returned. We do want to make sure we aren't over-specifying the result of the server reply for `explain` if the server api response is not under semver (I do no recall whether or not it is for explain), but I am not convinced that we have found the sweet spot between overspecifying and underspecifying it here. It may be worthwhile to take a step back and ask "what is the behavior that is important for the *driver* to guarantee here?" If it's to make sure that we correctly set `explain`, then we could just use command monitoring. If it's to make sure we don't strip explain data from the reply, we need to define what data we expect to see for each case, and then capture that in the tests.

Reply:
The shape of the explain result changes based on server version and internal implementation of the query engine, so not semver for driver nor server. I agree with testing commands sent, although, it's good to have _some_ verification that the explain result can be correctly returned from our APIs since they do not match the expected return type of those methods. The verification has to be some definition of lightly asserted since we can't rely on explain having the same shape all the time.

I do think its scope creep to add command monitoring tests since we set out to refactor what exists, and confirming queryPlanner on the resulting object seems like the sweet spot for the moment to confirm explain response can make it out of our APIs. @malikj2000 thoughts?

Reply:
I agree, I do think command monitoring tests would help make the tests much clearer and more accurate to what the tests were actually meant to do. However, I think this is a good stopping point for this PR since we are very close to being done. We should open up another ticket for adding command monitoring tests.

Reply:
You're right when it comes to scope creep on this ticket. @malikj2000 and/or @nbbeeken can you file a follow-up ticket to address this?

Reply:
Specifically capturing the changes we need to make to the explain tests to capture driver-specific behaviour with command monitoring.

In addition, if you could update the context blocks for `explain set to true` and `explain set to false` to note that these represent highest and lowest verbosity respectively, that would be great and I'd be clear to merge.

Reply:
Scope creep concerns aside, the tests right now are misleading. They claim that it's important to us to test that `queryPlanner` is returned, but that's not actually what the intention is. We should make sure our test case titles and assertions capture what's important. And adding command monitoring isn't anymore effort than filing a ticket that accurately describes the work that remains to be done. Yes, from an intern point of view, this refactor was an exercise in converting callbacks to async/await, but there's no point in shuffling these tests about if they aren't going to be useful to us as a team in the codebase going forward. I don't want to check in something that is going to create more work for us in the future (in the form of wasting more engineering time trying to figure out what the tests are or aren't supposed to test). When we discussed the test refactor project with Neal, we explicitly talked about the fact that the hardest part of working with legacy tests is figuring out what the tests are supposed to do and what the correct assertions should be. If we use `queryPlanner` as a sanity check to make sure that the explain portion of the response isn't stripped, that's fine, and we can leave that assertion in with a comment, but that can't be the main and only thing we test here. Ideally we would test the driver's branching logic for the handling of explain, which is different on an api by api basis.

Reply:
@dariakp You're right that the tests generally are either unclear or misleading, whether they use the new syntax or not. My bad for not being as discerning as I should have been with that; I was focused a bit too closely on the asyncifying part of the PR.

@malikj2000 let's go through the docs ([here](https://www.mongodb.com/docs/v6.0/reference/command/explain/#mongodb-dbcommand-dbcmd.explain) and [here](https://www.mongodb.com/docs/v6.0/reference/explain-results/)) to get a solid idea of what explain is supposed to do in the first place and get the tests to reflect that.

Reply:
test not passing on aggregate, so will go back to lines 231 on and see what aggregate returns and adjust testing based on that